src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -103,10 +109,14 @@ public function encode(Envelope $envelope): array
 
         $headers = ['type' => \get_class($envelope->getMessage())] + $this->encodeStamps($envelope) + $this->getContentTypeHeader();
 
-        return [
-            'body' => $this->serializer->serialize($envelope->getMessage(), $this->format, $context),
-            'headers' => $headers,
-        ];
+        try {
+            return [
+                'body' => $this->serializer->serialize($envelope->getMessage(), $this->format, $context),
+                'headers' => $headers,
+            ];
+        } catch (NotNormalizableValueException $e) {
+            throw new LogicException('No normalizers were found to encode message. Try running "composer require symfony/property-access".', $e->getCode(), $e);
+        }
```

Comment:
Under what situations would serialization fail if you're missing the component.

Reply:
The component is needed to serialize objects. Without it (and without a normalizer of your own that can handle objects), a `NotNormalizableValueException` will be thrown.